contracts/Depository.sol

```diff
@@ -315,9 +368,11 @@ contract Depository is IErrorsTokenomics {
         // Get the LP token address
         address token = product.token;
 
-        // Calculate the payout in OLAS tokens based on the LP pair with the discount factor (DF) calculation
+        // Calculate the payout in OLAS tokens based on the LP pair with the inverse discount factor (IDF) calculation
         // Note that payout cannot be zero since the price LP is non-zero, otherwise the product would not be created
-        payout = IGenericBondCalculator(bondCalculator).calculatePayoutOLAS(tokenAmount, product.priceLP);
+        payout = IBondCalculator(bondCalculator).calculatePayoutOLAS(tokenAmount, product.priceLP,
+            // Encode parameters required for the IDF calculation
+            abi.encode(msg.sender, bondVestingTime, productMaxVestingTime, supply, product.payout));
```

Comment:
abi.encode(msg.sender, bondVestingTime, productMaxVestingTime, supply, product.payout)
This design seems like overengineering to me.
better just
IBondCalculator(bondCalculator).calculatePayoutOLAS(tokenAmount, product.priceLP,
            // Encode parameters required for the IDF calculation
            msg.sender, bondVestingTime, productMaxVestingTime, supply, product.payout);